src/mongocxx/events/command_failed_event.hpp

```diff
@@ -16,6 +16,10 @@
 
 #include <memory>
 
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <bsoncxx/private/libbson.hh>
```

Comment:
The C++ driver cannot use private headers in public headers. libbson / libmongoc symbols should be an implementation detail and not exposed.

Reply:
Right, I have been wondering about this one (and I think I even commented somewhere!)-- I wasn't able to otherwise expose bson_oid_t. But, I see you've answered how this works in the C++ driver below! Thank you!